ninja-core/src/main/java/ninja/Bootstrap.java

```diff
@@ -0,0 +1,271 @@
+    static public final String ROUTES_CONVENTION_LOCATION = "conf.Routes";
+    static public final String NINJA_CONVENTION_LOCATION = "conf.Ninja";
+
+    private final NinjaPropertiesImpl ninjaProperties;
+    private final List<Module> modulesToLoad;
+    private final Optional<String> applicationModulesBasePackage;
+    private Injector injector = null;
+
+    public Bootstrap(NinjaPropertiesImpl ninjaProperties) {
+        Preconditions.checkNotNull(ninjaProperties);
+        this.ninjaProperties = ninjaProperties;
+        this.modulesToLoad =  new ArrayList<>();
+        
+        // custom base package for application modules (e.g. com.example.conf.Routes)
+        this.applicationModulesBasePackage
+            = Optional.fromNullable(ninjaProperties.get(
+                    NinjaConstant.APPLICATION_MODULES_BASE_PACKAGE));
+        
+    }
+
+    public synchronized void boot() {
+        // 1. initialize logging
+        initLogback();
+
+        if (this.injector != null) {
+            throw new RuntimeException("Bootstrap already booted");
+        }
+
+        // 2. configure all modules for injector
+        try {
+            configure();
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to configure Ninja", e);
+        }
+        
+        // 3. create injector
+        long startTime = System.currentTimeMillis();
+
+        try {
+            initInjector();
+        } catch (Exception e) {
+            throw new RuntimeException("Ninja injector cannot be generated. Please check log for further errors.", e);
+        }
+        
+        long injectorStartupTime = System.currentTimeMillis() - startTime;
+        logger.info("Ninja injector started in " + injectorStartupTime + " ms.");
+        
+        // 4. initialize routes
+        try {
+            initRoutes();
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to initialize Ninja routes", e);
+        }
+        
+        // 5. framework started!
+        Ninja ninja = injector.getInstance(Ninja.class);
+        ninja.onFrameworkStart();
+    }
+
+    public synchronized void shutdown() {
+        if (this.injector != null) {
+            Ninja ninja = injector.getInstance(Ninja.class);
+            ninja.onFrameworkShutdown();
+            injector = null;
+            ninja = null;
+        } else {
+            logger.error("Shutdown of Ninja not clean => injector already null.");
+        }
+    }
+    
+    public Injector getInjector() {
+        return this.injector;
+    }
+    
+    public void addModule(Module module) {
+        this.modulesToLoad.add(module);
+    }
+    
+    protected void configure() throws Exception {
+
+        // Bind lifecycle support
+        addModule(LifecycleSupport.getModule());
+        
+        // Scheduling support
+        addModule(SchedulerSupport.getModule());
+
+        // TODO: this is where problems start to occur...
```

Comment:
I was attempting to figure out how to build a "lite" version of Ninja -- just an accidental leftover comment that I'll remove.